### PR TITLE
fix(pt-expt): detach tensors before numpy serialization

### PR DESCRIPTION
**Pull request by njzjz**
## Summary
- detach torch tensors that require gradients before exporting them through `to_numpy_array`
- document the PyTorch `torch.asarray` behavior change from pytorch/pytorch@a97dcf9
- add a regression test for serializing a trainable `torch.nn.Parameter`

## Tests
- `ruff check deepmd/dpmodel/common.py source/tests/consistent/test_array_api.py`
- `ruff format --check deepmd/dpmodel/common.py source/tests/consistent/test_array_api.py`
- `pytest source/tests/consistent/test_array_api.py::TestArrayConversion::test_torch_parameter_requires_grad -q`
- `pytest source/tests/consistent/test_array_api.py -q`
- `srun --gres=gpu:1 dp --pt-expt train input.json --skip-neighbor-stat`
- `srun --gres=gpu:1 dp --pt-expt train input.json`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved conversion of machine learning tensors to NumPy arrays when gradient tracking is enabled, preventing conversion failures while maintaining correct values.

* **Tests**
  * Added coverage for converting gradient-enabled PyTorch parameters, including checks for output values, dtype, and correct behavior before/after conversion.

### CODE REVIEW

**Review by njzjz — COMMENTED** [review]
LGTM

— Opus 4.8

**Review by wanghan-iapcm — APPROVED** [review]
